### DISCUSSION TITLE
How to extract subtitles to a file by using mpv

### DISCUSSION THREAD

**Discussion by ahaoboy** [mpv-player/mpv#14942] [Q&A]
I'm trying to implement a subtitle translation script for mpv. I need a command similar to ```ffmpeg -i Movie.mkv -map 0:s:0 subs.srt```, or a script to get all the subtitle information of the current video.

I can get the current subtitles through the sub-text attribute, but due to network latency, the subtitles will be delayed. I want to try to translate all the subtitles in the video in advance, so that the subtitles and the video can match.

**Comment by guidocella** [comment]
mpv doesn't support subtitle extraction. You can call `subprocess` to get subtitles with `ffmpeg` like `select.lua` does.

**Reply by ahaoboy** [reply]
Using ffmpeg requires users to install ffmpeg, which is very troublesome for users who are not familiar with computers. If it is packaged into the program, it will also increase the size of the program.

**Comment by verygoodlee** [comment]
[sub_export.lua](https://github.com/dyphire/mpv-scripts/blob/main/sub_export.lua) might help you

**Reply by ahaoboy** [reply]
Still using ffmpeg
